docs/guides/server/db.adoc

```diff
@@ -358,6 +357,137 @@ In the same way the migrationExport to point to a specific file and location:
+var user = em.find(org.your.extension.UserEntity.class, 123L);
+----
+
+For more examples, look at the link:https://github.com/keycloak/keycloak-quickstarts/tree/main/extension/user-storage-jpa[Keycloak User Storage SPI Quickstart].
+
+=== Configuration via environment variables
+If you do not want to configure the datasource via CLI or `keycloak.conf` properties, you can use the environment variables.
+
+You can set the DB kind via environment variables (for the `user-store` datasource) as follows:
+
+[source,bash]
+----
+export KC_DB_KIND_USER_STORE=postgres
+export KC_DB_USERNAME_USER_STORE=my-username
+----
+
+It maps to the `db-kind-user-store` and `db-username-user-store` {project_name} properties due to the default mapping of the `\_` (underscore) to the `-` (dash) for environment variables.
+However, sometimes, the name of the datasource might contain some special characters like `_`, `$` or `.`
+
+In order to have it properly configured via the {project_name} environment variables, you need to explicitly say what the key for the datasource should look like.
+You can use a pair of unique {project_name} environment variables with a special case of the `KCKEY_`.
+
+For instance, for a datasource with the name __user_store$marketing__, you can set environment variables as follows:
+
+[source,bash]
+----
+export KC_USER_STORE_DB_KIND=mariadb
+export KCKEY_USER_STORE_DB_KIND=db-kind-user_store$marketing
+----
+
+You can find more information in the guide <@links.server id="configuration"/>, in subsection _Formats for environment variable keys with special characters_.
+
+=== Backward compatibility for the `quarkus.properties`
+In the past, we instructed users to use raw Quarkus properties to configure additional datasources in some places.
+However, as using Quarkus properties in the `conf/quarkus.properties` file is considered **unsupported**, it is strongly recommended to use the dedicated additional datasources options as described above.
+
+Before you are able to migrate to the dedicated options, you can still specify the datasource settings via the Quarkus properties as follows:
+
+[source,properties]
+----
+quarkus.datasource.user-store.db-kind=h2
+quarkus.datasource.user-store.username=sa
+quarkus.datasource.user-store.jdbc.url=jdbc:h2:mem:user-store;DB_CLOSE_DELAY=-1
+quarkus.datasource.user-store.jdbc.transactions=xa
+----
+
+WARNING: Use Quarkus properties **without quotation** for the datasource name, as properties with the quoted datasource name clash with the new datasource options mapping.
+Therefore, use `quarkus.datasource.user-store.db-kind=h2`, instead of `quarkus.datasource."user-store".db-kind=h2` to prevent any issues.
+
+<@opts.printRelevantOptions includedOptions="db db-* transaction-xa-enabled" excludedOptions="db-*-<datasource>">
+
+=== Additional datasources options
+<@opts.includeOptions includedOptions="*-<datasource>"/>
+
+</@opts.printRelevantOptions>
+
+
 </@tmpl.guide>
 
 == Configuring the connection pool
```

Comment:
Noticed one blocker. This should be probably placed above the Relevant options which should always be the last in a guide.